playground/backend/internal/environment/beam.go

```diff
@@ -15,14 +15,66 @@
 
 package environment
 
-import pb "beam.apache.org/playground/backend/internal/api"
+import (
+	pb "beam.apache.org/playground/backend/internal/api"
+	"github.com/tkanos/gonfig"
+	"os"
+	"strings"
+)
+
+// CmdConfig contains needed for compiling and execution of the code commands
+type CmdConfig struct {
+	CompileCmd  string   `json:"compile_cmd"`
+	RunCmd      string   `json:"run_cmd"`
+	CompileArgs []string `json:"compile_args"`
+	RunArgs     []string `json:"run_args"`
+}
 
 // BeamEnvs contains all environments related of ApacheBeam. These will use to run pipelines
 type BeamEnvs struct {
 	ApacheBeamSdk pb.Sdk
+	CmdConfig     *CmdConfig
 }
 
 // NewBeamEnvs is a BeamEnvs constructor
 func NewBeamEnvs(apacheBeamSdk pb.Sdk) *BeamEnvs {
-	return &BeamEnvs{ApacheBeamSdk: apacheBeamSdk}
+	config := readConfig(apacheBeamSdk)
+	return &BeamEnvs{ApacheBeamSdk: apacheBeamSdk, CmdConfig: config}
+}
+
+func readConfig(apacheBeamSdk pb.Sdk) *CmdConfig {
```

Comment:
I mentioned before, that all functionality to extract envs from os envs should be implemented in environment service.

Reply:
this structure is DTO